site/docs/spec.md

```diff
@@ -480,6 +499,8 @@ Table metadata consists of the following fields:
 | _optional_ | _optional_ | **`current-snapshot-id`**| `long` ID of the current table snapshot. |
 | _optional_ | _optional_ | **`snapshots`**| A list of valid snapshots. Valid snapshots are snapshots for which all data files exist in the file system. A data file must not be deleted from the file system until the last snapshot in which it was listed is garbage collected. |
 | _optional_ | _optional_ | **`snapshot-log`**| A list (optional) of timestamp and snapshot ID pairs that encodes changes to the current snapshot for the table. Each time the current-snapshot-id is changed, a new entry should be added with the last-updated-ms and the new current-snapshot-id. When snapshots are expired from the list of valid snapshots, all entries before a snapshot that has expired should be removed. |
+| _optional_ | _optional_ | **`sort-orders`**| A list of sort orders, stored as full sort order objects. |
+| _optional_ | _optional_ | **`default-sort-order-id`**| Default sort order id of the table. Note that this could be used by writers, but is not used when reading because reads use the specs stored in manifest files. |
```

Comment:
@aokolnychyi, do you think that sort order information should be optional in v2?

Reply:
I think that we need to clarify this a bit more. Sort order is _optional_ in v2 because v1 metadata is still valid. But a v2 writer must always write the `sort-orders` and `default-sort-order-id` fields. I think we need to change this to _required_ for v2 and make sure there is text stating that these requirements are for writers and that v2 readers must accept all valid v1 metadata.

Reply:
I think the last sentence in the current [formatv2 section](https://iceberg.apache.org/spec/#version-2) basically coverers the v2 readers accepting valid v1 metadata part. I'll add `sort-orders` and `default-sort-order-id` to the list of this section for now, and do you think if we should move this section/add more explanation about this in other places?

Reply:
Yes, since that section is intended to clarify, not specify. That doesn't need to be part of this PR, though.